js/utils/README.md

```diff
@@ -1,75 +1,231 @@
+
+    There will be 4 arrays named string (two in an if statement, rest two in the else statement).
+    Add your theme's name to the bottom of the array
+
+    ```javascript
+    string = [[...],
+        ["light", _("Light Mode"), "innerHTML"],
+        ["dark", _("Dark Mode"), "innerHTML"],
+        ["pastel", _("Pastel Theme"), "innerHTML"],
+        ["your_theme", _("Your Theme's Name"), "innerHTML"]];
+    ```
+
+    ```javascript
+    string = [_(...),
+    _("Light Mode"),
+    _("Dark Mode"),
+    _("Pastel Theme"),
+    _("Your Theme Name")];
+    ```
+
+    There will be two same arrays in the else statement, repeat the process. This is to display your theme's name in the dropdown menu item you created in Step 1.
+
+3.  **Understanding the process a little bit**
+
+    There is a renderThemeSelectIcon function in js/toolbar.js
+
+    ```javascript
+    renderThemeSelectIcon(themeBox, themes) {
+        const themeSelectIcon = docById("themeSelectIcon");
+        let themeList = themes;
+        themeSelectIcon.onclick = () => {
+            themeList.forEach((theme) => {
+                docById(theme).onclick = () => themeBox[`${theme}_onclick`](this.activity);
+            });
+        };
+    }
+    ```
+
+    and it is called in js/activity.js
+
+    ```javascript
+    this.toolbar.renderThemeSelectIcon(this.themeBox, this.themes);
+    ```
+
+    This adds functionality to our options of themes using the themeBox object but leave themeBox for now. Let's get to the this.themes part which is an array in the Activity class in js/activity.js.
+
+    ```javascript
+    class Activity {
+        constructor(
+            ...
+            this.themes = ["light", "pastel", "dark", "your_theme"];
+            ...
+        )
+    }
+    ```
+
+    Add your theme here too. In activity.js, you will find just below there is this for loop, this checks for all the themes in the themes array we just saw above and then add the theme class in themes.css (which you will add in later steps) which matches the themePreference object in localStorage.
+
+    ```javascript
+    for (let i = 0; i < this.themes.length; i++) {
+        if (this.themes[i] === this.storage.themePreference) {
+            body.classList.add(this.themes[i]);
         } else {
-            this.storage.myThemeName = "darkMode";
-        }
-        try {
-            window.location.reload();
-        } catch (e) {
-            console.error("Error reloading the window:", e);
+            body.classList.remove(this.themes[i]);
         }
-    };
-   ```
+    }
+    ```
 
-5. **Applying the Theme After Reload**
+    If no theme is selected, i.e., there is no themePreference in localStorage. The styles will default to light mode, as it is the base styling without a class in themes.css. But when there is a themePreference in localStorage, the class respective to the chosen theme will be added to the elements which will apply because of higher specificity.
```

Comment:
If no theme has been selected (i.e., themePreference has not been defined in localStorage), the styles will default to "light". Otherwise, the theme defined in themePreference will be applied.